gateway_tests/conftest.py

```diff
@@ -434,3 +617,230 @@ def compare_structures(gw_struct, ioc_struct) -> str:
                 f"Element '{key}' : GW has '{gateway_value}', IOC has '{ioc_value}'"
             )
     return "\n\t".join(differences)
+
+
+@contextlib.contextmanager
+def ca_subscription(
+    pvname: str,
+    callback: PyepicsCallback,
+    mask: int = epics.dbr.DBE_VALUE,
+    form: str = "time",
+    count: int = 0,
+    timeout: float = 0.5,
+) -> ContextManager[int]:
+    """
+    Create a low-level channel and subscription for a provided pvname.
+
+    Yields channel identifier.
+
+    Parameters
+    ----------
+    pvname : str
+        The PV name suffix, not including "ioc:" or "gateway:".
+
+    callback : callable
+        A callback function to use for value updates.
+
+    mask : int, optional
+        The DBE mask to use for subscriptions.
+
+    form : {"native", "time", "ctrl"}, optional
+        The form to request.
+
+    count : int, optional
+        The number of elements to request.
+
+    timeout : float, optional
+        The timeout in seconds for connection.
+
+    Yields
+    ------
+    channel : int
+        The Channel Access client channel ID.
+    """
+    event_id = None
+    chid = epics.ca.create_channel(pvname)
+    try:
+        connected = epics.ca.connect_channel(chid, timeout=timeout)
+        assert connected, f"Could not connect to channel: {pvname}"
+
+        (_, _, event_id) = epics.ca.create_subscription(
+            chid,
+            mask=mask,
+            use_time=form == "time",
+            use_ctrl=form == "ctrl",
+            callback=callback,
+            count=count,
+        )
+        yield chid
+    finally:
+        if event_id is not None:
+            epics.ca.clear_subscription(event_id)
+        epics.ca.clear_channel(chid)
+
+
+@contextlib.contextmanager
+def ca_subscription_pair(
+    pvname: str,
+    ioc_callback: PyepicsCallback,
+    gateway_callback: PyepicsCallback,
+    mask: int = epics.dbr.DBE_VALUE,
+    form: str = "time",
+    count: int = 0,
+    timeout: float = 0.5,
+    ioc_prefix: str = "ioc:",
+    gateway_prefix: str = "gateway:",
+) -> ContextManager[tuple[int, int]]:
+    """
+    Create low-level channels + subscriptions for IOC and gateway PVs.
```

Comment:
This seems like a crazy helpful context manager

Reply:
It sure reduces the verbosity of the boilerplate in tests, at least!